internal/mapping.go

```diff
@@ -141,26 +141,45 @@ func GetSpannerCols(conv *Conv, srcTable string, srcCols []string) ([]string, er
 // (across the database). But in some source databases, such as PostgreSQL,
 // they only have to be unique for a table. Hence we must map each source
 // constraint name to a unique spanner constraint name.
-func GetSpannerKeyName(srcKeyName string, schemaForeignKeys map[string]bool) string {
+func ToSpannerForeignKey(srcKeyName string, schemaForeignKeys map[string]bool) string {
 	if srcKeyName == "" {
 		return ""
 	}
+	return getUniqueId(srcKeyName, schemaForeignKeys)
+}
+
+// ToSpannerIndexKey maps source index key name to
+// legal Spanner index key name
+// We need to make sure
+// of the following things:
+// a) the new index key name is legal
+// b) the new index key name doesn't clash with other Spanner
+//    index key names
+// Note that index key constraint names in Spanner have to be globally unique
+// (across the database). But in some source databases, such as MySQL,
+// they only have to be unique for a table. Hence we must map each source
+// constraint name to a unique spanner constraint name.
+func ToSpannerIndexKey(srcKeyName string, schemaIndexKeys map[string]bool) string {
+	return getUniqueId(srcKeyName, schemaIndexKeys)
+}
+
+func getUniqueId(srcKeyName string, schemaKeys map[string]bool) string {
```

Comment:
Thanks for abstracting this functionality out into a separate function!
I think id is the right concept to use here.

Suggest some slight tweaking:
getUniqueId -> getSpannerId  (since it both maps to spanner and manages uniqueness)
srcKeyName -> srcId
schemaKeys -> used

and then in the code, systematically use Id for KeyName.